File: [2018-07-20_2018-07-25]angular/code/employeApp/src/app/metier/employe.ts
export class Employe {
    public constructor(public id: number,
                        public nom: string,
                        public prenom: string,
                        public email: string,
                        public salaire: number) {}
    
    public copyFrom(source : Employe) {
        this.id = source.id;
        this.nom = source.nom;
        this.prenom = source.prenom;
        this.email = source.email;
        this.salaire = source.salaire;
    }

}

File: [2018-07-26_2018-08-08]java8/code/heritageSimpleForm/src/heritageSimpleForm/metier/Employe.java
package heritageSimpleForm.metier;

import java.time.LocalDate;

public class Employe extends Personne {
	private String poste;
	private double salaire;
	
	public String getPoste() {return poste;}
	public void setPoste(String poste) {this.poste = poste;}
	public double getSalaire() {return salaire;}
	public void setSalaire(double salaire) {this.salaire = salaire;}
	
	public Employe() {}
	public Employe(String nom, String prenom, LocalDate dateNaissance, String poste, double salaire) {
		super(nom, prenom, dateNaissance);
		this.poste = poste;
		this.salaire = salaire;
	}
	
	@Override
	public String toString() {
		//return super.super.
		return "Employe [poste=" + poste + ", salaire=" + salaire + ", getNom()=" + getNom() + ", getPrenom()="
				+ getPrenom() + ", getDateNaissance()=" + getDateNaissance() + "]";
	}
	
	@Override
	public String toCsvLine() {
		return super.toCsvLine() + "," + this.poste + "," + this.salaire;
	}

	
	
	
	
	
	
}
